<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AisParserMainWindow</class>
 <widget class="QMainWindow" name="AisParserMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>599</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>564</height>
   </size>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>ICODE-MDA AIS Parser</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="inputFrame">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>260</y>
      <width>381</width>
      <height>221</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="filenameLineEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>261</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QComboBox" name="inputTypeComboBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>111</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Satellite AIS Log</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Terrestrial AIS Log</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Satellite AIS TCP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Terrestrial AIS TCP</string>
      </property>
     </item>
    </widget>
    <widget class="QToolButton" name="filenameToolButton">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>90</y>
       <width>25</width>
       <height>19</height>
      </rect>
     </property>
     <property name="text">
      <string>...</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="portSpinBox">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>10000</number>
     </property>
    </widget>
    <widget class="QLabel" name="inputTypeLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Type</string>
     </property>
    </widget>
    <widget class="QLabel" name="filenameLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Filename</string>
     </property>
    </widget>
    <widget class="QLabel" name="portLabel">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Port</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="inputHostnameLineEdit">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>261</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>localhost</string>
     </property>
    </widget>
    <widget class="QLabel" name="inputHostnameLabel">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Hostname</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="outputFrame">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>30</y>
      <width>301</width>
      <height>481</height>
     </rect>
    </property>
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <widget class="QLineEdit" name="outputDirectoryLineEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>191</width>
       <height>20</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QComboBox" name="outputTypeComboBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Tab-Seperated Files</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Comma-Seperated Files</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>PostgreSQL Database</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>MySql Database</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Oracle Database</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>KML Placemarks</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>KML Tracks</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ESRI Shapefile Points</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ESRI Shapefile Tracks</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="outputTypeLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Type</string>
     </property>
    </widget>
    <widget class="QToolButton" name="outputDirectoryToolButton">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>90</y>
       <width>25</width>
       <height>19</height>
      </rect>
     </property>
     <property name="text">
      <string>...</string>
     </property>
    </widget>
    <widget class="QLabel" name="outputDirectoryLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Output Directory</string>
     </property>
    </widget>
    <widget class="QFrame" name="databaseInformationFrame">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>241</width>
       <height>331</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="usernameLabel">
        <property name="text">
         <string>Username</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="usernameLineEdit">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="passwordLabel">
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="passwordLineEdit">
        <property name="text">
         <string/>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="hostnameLabel">
        <property name="text">
         <string>Hostname</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="hostnameLineEdit">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="databaseNameLabel">
        <property name="text">
         <string>Database Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="databaseNameLineEdit">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="tableNameLabel">
        <property name="text">
         <string>Dynamic Table Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="tableNameLineEdit">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="staticTableCheckBox">
        <property name="text">
         <string>Push static messages to separate table</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="staticTableNameLabel">
        <property name="text">
         <string>Static Table Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="staticTableNameLineEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="numIterationsLabel">
        <property name="text">
         <string>Number of Iterations</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="numIterationsSpinBox">
        <property name="maximum">
         <number>1000000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="databaseInformationLabel">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Database Information</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="messagesPerFileSpinBox">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000000</number>
     </property>
    </widget>
    <widget class="QLabel" name="messagesPerFilelabel">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>10</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Messages Per File</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="instructionsLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Instructions</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="instructionTextBrowser">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>381</width>
      <height>192</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Select an input type and specify the additional options in the Input window. Then select an output type and specify the additional options in the Output window.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;&lt;br /&gt;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;Note:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;The difference between the satellite AIS and terrestrial AIS logfiles is the number of columns the data has. The satellite AIS data has an extra column of data where the STREAMID is placed that indicates which satellite/vendor provided the posit.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="inputLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>240</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Input</string>
    </property>
   </widget>
   <widget class="QLabel" name="outputLabel">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>10</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Output</string>
    </property>
   </widget>
   <widget class="QPushButton" name="startPushButton">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>520</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopPushButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>520</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionLicense"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
  </widget>
  <action name="actionClose">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionLicense">
   <property name="text">
    <string>License</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>instructionTextBrowser</tabstop>
  <tabstop>inputTypeComboBox</tabstop>
  <tabstop>filenameLineEdit</tabstop>
  <tabstop>filenameToolButton</tabstop>
  <tabstop>inputHostnameLineEdit</tabstop>
  <tabstop>portSpinBox</tabstop>
  <tabstop>outputTypeComboBox</tabstop>
  <tabstop>messagesPerFileSpinBox</tabstop>
  <tabstop>outputDirectoryLineEdit</tabstop>
  <tabstop>outputDirectoryToolButton</tabstop>
  <tabstop>usernameLineEdit</tabstop>
  <tabstop>passwordLineEdit</tabstop>
  <tabstop>hostnameLineEdit</tabstop>
  <tabstop>databaseNameLineEdit</tabstop>
  <tabstop>tableNameLineEdit</tabstop>
  <tabstop>staticTableCheckBox</tabstop>
  <tabstop>staticTableNameLineEdit</tabstop>
  <tabstop>numIterationsSpinBox</tabstop>
  <tabstop>startPushButton</tabstop>
  <tabstop>stopPushButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>outputTypeComboBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>manageOutputVisibility(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>443</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputTypeComboBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>manageInputVisibility(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startPushButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>startParsingAis()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>524</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopPushButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>stopParsingAis()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>632</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>667</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filenameToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>fileChooser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>315</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputDirectoryToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>directoryChooser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>697</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>782</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>staticTableCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>AisParserMainWindow</receiver>
   <slot>manageStaticVisibility(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>559</x>
     <y>419</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startParsingAis()</slot>
  <slot>stopParsingAis()</slot>
  <slot>manageOutputVisibility(QString)</slot>
  <slot>manageInputVisibility(QString)</slot>
  <slot>fileChooser()</slot>
  <slot>directoryChooser()</slot>
  <slot>manageStaticVisibility(int)</slot>
 </slots>
</ui>
